lib/loggers/index.js

```diff
@@ -51,31 +52,42 @@ const container = new winston.Container();
 // override get to use a wrapper so loggers that are retrieved via `.get()`
 // prior to logger initialization will receive the updated configuration
 // when logging after initialization
-container._get = container.get;
+// container._get = container.get;
 container._add = container.add;
-container.get = container.add = function(id) {
-  const existing = container.loggers[id];
-  let logger = container._get.apply(container, arguments);
+container.add = function(id) {
+  const existing = container.has(id);
+  let logger = container._add.apply(container, arguments);
   if(!existing) {
     const wrapper = Object.create(logger);
-    wrapper.log = function(level, msg, meta) {
-      // merge per-logger and per-log meta and call parent logger
-      meta = Object.assign({}, this.meta, meta);
-      return Object.getPrototypeOf(wrapper).log.apply(
-        wrapper, [level, msg, meta]);
+    wrapper.log = function(level, msg, ...meta) {
+      // get default meta to ensure it is always applied
+      const {defaultMeta} = this;
+      // use `Object.getPrototypeOf` because the prototype may change
+      // while the logging subsystem initializes
+      const log = Object.getPrototypeOf(wrapper).log;
+      if(arguments.length === 1) {
+        return log.apply(wrapper, [{...defaultMeta, ...level}]);
+      }
+      if(meta.length === 0) {
+        meta.push({defaultMeta});
+      } else {
+        meta[0] = {...defaultMeta, ...meta[0]};
+      }
+      return log.apply(wrapper, [level, msg, ...meta]);
     };
-    wrapper.child = function(childMeta) {
+    const createChild = wrapper.child;
+    wrapper.child = function(childMeta, ...args) {
       // simple string name is shortcut for {module: name}
       if(typeof childMeta === 'string') {
         childMeta = {module: childMeta};
       }
-      // create child logger with merged meta data from parent
-      const child = Object.create(this);
-      child.meta = Object.assign({}, this.meta, childMeta);
-      child.setLevels(levels);
+      // create child logger from wrapper and merge child meta into parent meta
+      const child = createChild.apply(wrapper, [childMeta, ...args]);
```

Comment:
@dlongley createChild does this which I think is just weird.  I don't think we need it.  I think making a copy and setting `defaultData` is adequate.  What do you think?

https://github.com/winstonjs/winston/blob/master/lib/winston/logger.js#L45-L71

Reply:
Yeah, I saw that it was doing special things with some `write` function, and I didn't know where/how it was used. That's why I kept it in there, to minimize changes. Presumably, `write` is used somewhere ... and it may break if we don't just keep the default behavior. I don't think it really hurts us to keep it.

Reply:
OK, the `.child` implementation (referenced above) accepts only one parameter.  I'll be removing the `...args` bit.